include/ginkgo/core/base/math.hpp

```diff
@@ -495,6 +495,16 @@ template <typename T>
 using next_precision = typename detail::next_precision_impl<T>::type;
 
 
+/**
+ * Obtains the previous type in the singly-linked precision list.
+ *
+ * @note Currently our lists contains only two elements, so this is the same as
+ *       next_precision.
+ */
+template <typename T>
+using previous_precision = next_precision<T>;
```

Comment:
Where is this used? I must have missed it

Reply:
From what I can tell, it is only used here: https://github.com/ginkgo-project/ginkgo/pull/1171/files#diff-e56543c2eafac5af3339aac83474d0ee4836b912a212c020817bc4742312ed70R524
(in `conversion_target_helper` in `include/ginkgo/core/distributed/vector.hpp`).
I am not sure if it is required though, but that is a different comment of mine.

Reply:
It's used in the conversion helper struct. The template parameter for that struct is the vector in higher precision and the input parameter is the one in lower precision, so I found it odd to use `next_precision` to get the type for the lower precision vector.